lib/spack/spack/test/cmd/dev_build.py

```diff
@@ -20,8 +22,8 @@
 pytestmark = pytest.mark.not_on_windows("does not run on windows")
 
 
-def test_dev_build_basics(tmpdir, mock_packages, install_mockery):
-    spec = spack.spec.Spec("dev-build-test-install@0.0.0 dev_path=%s" % tmpdir)
+def test_dev_build_basics(tmpdir, install_mockery):
```

Comment:
(no changes requested) This is fine, but in general we should start converting from the deprecated `tmpdir` to `tmp_path`. See https://docs.pytest.org/en/7.4.x/how-to/tmp_path.html#tmpdir-and-tmpdir-factory

Reply:
If it had the same features, yes ;p